jax/_src/sourcemap.py

```diff
@@ -0,0 +1,215 @@
+# Copyright 2024 The JAX Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from dataclasses import dataclass
+from typing import Iterable, Union
+
+### An implementation of sourcemaps following TC39 (see https://tc39.es/source-map/)
+
+# Each segment is made up of 1, 4 or 5 variable-length fields
+# (see https://tc39.es/source-map/#mappings-structure).
+Segment = Union[tuple[int], tuple[int, int, int, int], tuple[int, int, int, int, int]]
+
+@dataclass(frozen=True)
+class SourceMap:
+    version: int
+    # file: str
+    # source_root: str
+    sources: list[str]
```

Comment:
Consider using `Sequence` instead of `list` here and elsewhere, unless we need these fields to be mutable (in which case `MutableSequence` would make that more clear).